Peer Review : Linux

## Testing using the Kafka producer client
I got this warning when running `mvn liberty:dev` from `start/system`
> `[WARNING] /Users/kartiksaravanaibm.com/workspace/liberty/draft-guide-reactive-service-testing/start/system/src/main/java/io/openliberty/guides/system/SystemService.java:[51,35] package osMean does not exist`

And these errors (right after the "server has started" message):
```
/root/workspace/draft-guide-reactive-service-testing/start/system/src/main/java/io/openliberty/guides/system/SystemService.java:51: error: package osMean does not exist
                        new osMean.getSystemLoadAverage())));
                                  ^
1 error
[INFO] Source compilation had errors.
```

```
[INFO] [err] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[INFO] [err] SLF4J: Defaulting to no-operation (NOP) logger implementation
[INFO] [err] SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

### Running the tests
When I ran the test through `mvn liberty:dev` from `start/system`, the test failed. The test passed when ran through `mvn liberty:dev` from `inventory/system`. Logs:

```
0 INFO org.microshed.testing.testcontainers.ApplicationContainer  - Discovered ServerAdapter: class org.testcontainers.containers.liberty.LibertyAdapter
12 INFO org.microshed.testing.testcontainers.ApplicationContainer  - Using ServerAdapter: org.testcontainers.containers.liberty.LibertyAdapter
23 INFO org.microshed.testing.jupiter.MicroShedTestExtension  - Using ApplicationEnvironment class: org.microshed.testing.testcontainers.config.HollowTestcontainersConfiguration
50 INFO org.testcontainers.containers.liberty.LibertyAdapter  - Auto-configuring ApplicationContainer instances with MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS=localhost:9093
1250 INFO org.testcontainers.dockerclient.DockerClientProviderStrategy  - Loaded org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy from ~/.testcontainers.properties, will try it first
3482 INFO org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy  - Found docker client settings from environment
3482 INFO org.testcontainers.dockerclient.DockerClientProviderStrategy  - Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=unix:///var/run/docker.sock
3972 INFO org.testcontainers.DockerClientFactory  - Docker host IP address is localhost
4077 INFO org.testcontainers.DockerClientFactory  - Connected to docker: 
  Server Version: 19.03.8
  API Version: 1.40
  Operating System: Ubuntu 18.04.4 LTS
  Total Memory: 1992 MB
6986 INFO org.testcontainers.DockerClientFactory  - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
6987 INFO org.testcontainers.DockerClientFactory  - Checking the system...
6989 INFO org.testcontainers.DockerClientFactory  - ✔︎ Docker server version should be at least 1.6.0
7694 INFO org.testcontainers.DockerClientFactory  - ✔︎ Docker environment should have more than 2GB free disk space
7839 INFO org.microshed.testing.testcontainers.config.HollowTestcontainersConfiguration  - Exposing fixed port 9093 for container confluentinc/cp-kafka:5.2.1
7845 INFO org.microshed.testing.testcontainers.config.HollowTestcontainersConfiguration  - Exposing fixed port 9083 for container HollowApplicationContainer
7851 INFO org.microshed.testing.testcontainers.config.TestcontainersConfiguration  - Starting containers [GenericContainer(exposedPorts=[9093], portBindings=[9093:9093/tcp], extraHosts=[], networkMode=null, network=org.testcontainers.containers.Network$NetworkImpl@636248f4, networkAliases=[tc-LXbTimAe], image=RemoteDockerImage(imageNameFuture=java.util.concurrent.CompletableFuture@f28160d7[Completed normally], imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE), env=[KAFKA_LOG_FLUSH_INTERVAL_MESSAGES=9223372036854775807, KAFKA_BROKER_ID=1, KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS=1, KAFKA_LISTENERS=PLAINTEXT://0.0.0.0:9093,BROKER://0.0.0.0:9092, KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1, KAFKA_INTER_BROKER_LISTENER_NAME=BROKER, KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT, KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0], labels={}, commandParts=[], binds=[], privilegedMode=false, volumesFroms=[], linkedContainers={}, startupCheckStrategy=org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy@dd94667f, startupAttempts=1, workingDirectory=null, shmSize=null, copyToFileContainerPathMap={}, dependencies=[], dockerClient=LazyDockerClient.INSTANCE, dockerDaemonInfo=null, containerId=null, containerName=null, containerInfo=null, waitStrategy=org.testcontainers.containers.wait.HostPortWaitStrategy@b860a4ad, logConsumers=[], createContainerCmdModifiers=[], tmpFsMapping=null, shouldBeReused=false), GenericContainer(exposedPorts=[9083], portBindings=[9083:9083/tcp], extraHosts=[], networkMode=null, network=org.testcontainers.containers.Network$NetworkImpl@636248f4, networkAliases=[tc-446F7PA7], image=RemoteDockerImage(imageNameFuture=org.testcontainers.images.RemoteDockerImage$1@c3c7d689, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE), env=[mp_jwt_verify_issuer=http://testissuer.com, MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS=localhost:9093, mp_jwt_verify_publickey=-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAm+GdjH3ykpSFldpHf4/v
h+VCvMyjOyAv16c2lRN2L1USh+QhgMIfW4F5zZ7FekGV5MXWR7+LaSe9wn2zzxb6
ePWlcSjpwexTRdlSPJBoi3a2jMPlZ9XdHU4q3fD916iQQYvzxeJsCJzy79la56A4
fwY4H86c32fqn6wJHeb+wwKU1ESnkhFMBn6aVfAtW/VNfaINVd0XZUAjcPW9aYUL
Fqy/pQJh/w+33pzGhFITt/+7XfaXF2fQDW5LehGBnQsbfg4xTH2tPlNZnsPoIu+j
RUPLTkd7KjyL2H6VCMLO/hWaEm8LsFnGxd+3+jbrBXBWttAqmsu3b1YQ5bLzkKlg
owIDAQAB
-----END PUBLIC KEY-----], labels={}, commandParts=[], binds=[], privilegedMode=false, volumesFroms=[], linkedContainers={}, startupCheckStrategy=org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy@750d2db3, startupAttempts=1, workingDirectory=null, shmSize=null, copyToFileContainerPathMap={}, dependencies=[GenericContainer(exposedPorts=[9093], portBindings=[9093:9093/tcp], extraHosts=[], networkMode=null, network=org.testcontainers.containers.Network$NetworkImpl@636248f4, networkAliases=[tc-LXbTimAe], image=RemoteDockerImage(imageNameFuture=java.util.concurrent.CompletableFuture@f28160d7[Completed normally], imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE), env=[KAFKA_LOG_FLUSH_INTERVAL_MESSAGES=9223372036854775807, KAFKA_BROKER_ID=1, KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS=1, KAFKA_LISTENERS=PLAINTEXT://0.0.0.0:9093,BROKER://0.0.0.0:9092, KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1, KAFKA_INTER_BROKER_LISTENER_NAME=BROKER, KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT, KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0], labels={}, commandParts=[], binds=[], privilegedMode=false, volumesFroms=[], linkedContainers={}, startupCheckStrategy=org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy@dd94667f, startupAttempts=1, workingDirectory=null, shmSize=null, copyToFileContainerPathMap={}, dependencies=[], dockerClient=LazyDockerClient.INSTANCE, dockerDaemonInfo=null, containerId=null, containerName=null, containerInfo=null, waitStrategy=org.testcontainers.containers.wait.HostPortWaitStrategy@b860a4ad, logConsumers=[], createContainerCmdModifiers=[], tmpFsMapping=null, shouldBeReused=false)], dockerClient=LazyDockerClient.INSTANCE, dockerDaemonInfo=null, containerId=null, containerName=null, containerInfo=HollowContainerInspection[app=GenericContainer(exposedPorts=[9083], portBindings=[9083:9083/tcp], extraHosts=[], networkMode=null, network=org.testcontainers.containers.Network$NetworkImpl@636248f4, networkAliases=[tc-446F7PA7], image=RemoteDockerImage(imageNameFuture=org.testcontainers.images.RemoteDockerImage$1@c3c7d689, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE), env=[mp_jwt_verify_issuer=http://testissuer.com, MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS=localhost:9093, mp_jwt_verify_publickey=-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAm+GdjH3ykpSFldpHf4/v
h+VCvMyjOyAv16c2lRN2L1USh+QhgMIfW4F5zZ7FekGV5MXWR7+LaSe9wn2zzxb6
ePWlcSjpwexTRdlSPJBoi3a2jMPlZ9XdHU4q3fD916iQQYvzxeJsCJzy79la56A4
fwY4H86c32fqn6wJHeb+wwKU1ESnkhFMBn6aVfAtW/VNfaINVd0XZUAjcPW9aYUL
Fqy/pQJh/w+33pzGhFITt/+7XfaXF2fQDW5LehGBnQsbfg4xTH2tPlNZnsPoIu+j
RUPLTkd7KjyL2H6VCMLO/hWaEm8LsFnGxd+3+jbrBXBWttAqmsu3b1YQ5bLzkKlg
owIDAQAB
-----END PUBLIC KEY-----], labels={}, commandParts=[], binds=[], privilegedMode=false, volumesFroms=[], linkedContainers={}, startupCheckStrategy=org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy@750d2db3, startupAttempts=1, workingDirectory=null, shmSize=null, copyToFileContainerPathMap={}, dependencies=[GenericContainer(exposedPorts=[9093], portBindings=[9093:9093/tcp], extraHosts=[], networkMode=null, network=org.testcontainers.containers.Network$NetworkImpl@636248f4, networkAliases=[tc-LXbTimAe], image=RemoteDockerImage(imageNameFuture=java.util.concurrent.CompletableFuture@f28160d7[Completed normally], imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE), env=[KAFKA_LOG_FLUSH_INTERVAL_MESSAGES=9223372036854775807, KAFKA_BROKER_ID=1, KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS=1, KAFKA_LISTENERS=PLAINTEXT://0.0.0.0:9093,BROKER://0.0.0.0:9092, KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1, KAFKA_INTER_BROKER_LISTENER_NAME=BROKER, KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT, KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0], labels={}, commandParts=[], binds=[], privilegedMode=false, volumesFroms=[], linkedContainers={}, startupCheckStrategy=org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy@dd94667f, startupAttempts=1, workingDirectory=null, shmSize=null, copyToFileContainerPathMap={}, dependencies=[], dockerClient=LazyDockerClient.INSTANCE, dockerDaemonInfo=null, containerId=null, containerName=null, containerInfo=null, waitStrategy=org.testcontainers.containers.wait.HostPortWaitStrategy@b860a4ad, logConsumers=[], createContainerCmdModifiers=[], tmpFsMapping=null, shouldBeReused=false)], dockerClient=LazyDockerClient.INSTANCE, dockerDaemonInfo=null, containerId=null, containerName=null, containerInfo=HollowContainerInspection[app=org.microshed.testing.testcontainers.ApplicationContainer@4494045f,args=<null>,config=<null>,created=<null>,driver=<null>,execDriver=<null>,hostConfig=<null>,hostnamePath=<null>,hostsPath=<null>,logPath=<null>,id=<null>,sizeRootFs=<null>,imageId=<null>,mountLabel=<null>,name=<null>,restartCount=<null>,networkSettings=<null>,path=<null>,processLabel=<null>,resolvConfPath=<null>,execIds=<null>,state=<null>,volumes=<null>,volumesRW=<null>,node=<null>,mounts=<null>,graphDriver=<null>,platform=<null>], waitStrategy=org.testcontainers.containers.wait.strategy.HttpWaitStrategy@3c51154b, logConsumers=[org.testcontainers.containers.output.Slf4jLogConsumer@748d16fd], createContainerCmdModifiers=[], tmpFsMapping=null, shouldBeReused=false),args=<null>,config=<null>,created=<null>,driver=<null>,execDriver=<null>,hostConfig=<null>,hostnamePath=<null>,hostsPath=<null>,logPath=<null>,id=<null>,sizeRootFs=<null>,imageId=<null>,mountLabel=<null>,name=<null>,restartCount=<null>,networkSettings=<null>,path=<null>,processLabel=<null>,resolvConfPath=<null>,execIds=<null>,state=<null>,volumes=<null>,volumesRW=<null>,node=<null>,mounts=<null>,graphDriver=<null>,platform=<null>], waitStrategy=org.testcontainers.containers.wait.strategy.HttpWaitStrategy@3c51154b, logConsumers=[org.testcontainers.containers.output.Slf4jLogConsumer@748d16fd], createContainerCmdModifiers=[], tmpFsMapping=null, shouldBeReused=false)] in parallel for class it.io.openliberty.guides.system.SystemServiceIT
7854 INFO org.microshed.testing.testcontainers.config.TestcontainersConfiguration  -   RemoteDockerImage(imageNameFuture=java.util.concurrent.CompletableFuture@f28160d7[Completed normally], imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE)
7855 INFO org.microshed.testing.testcontainers.config.TestcontainersConfiguration  -   RemoteDockerImage(imageNameFuture=org.testcontainers.images.RemoteDockerImage$1@c3c7d689, imagePullPolicy=DefaultPullPolicy(), dockerClient=LazyDockerClient.INSTANCE)
7870 INFO 🐳 [confluentinc/cp-kafka:5.2.1]  - Creating container for image: confluentinc/cp-kafka:5.2.1
8373 INFO 🐳 [confluentinc/cp-kafka:5.2.1]  - Starting container with ID: bc8b7dd7074649141eedc9e518f38727be8cdd3631d778829c2b22e26484e7c7
10616 INFO 🐳 [confluentinc/cp-kafka:5.2.1]  - Container confluentinc/cp-kafka:5.2.1 is starting: bc8b7dd7074649141eedc9e518f38727be8cdd3631d778829c2b22e26484e7c7
31739 INFO 🐳 [confluentinc/cp-kafka:5.2.1]  - Container confluentinc/cp-kafka:5.2.1 started in PT23.87304143S
32269 INFO org.testcontainers.containers.wait.strategy.HttpWaitStrategy  - HollowApplicationContainer: Waiting for 30 seconds for URL: http://9.46.85.200:9083/health/ready
[INFO] [AUDIT   ] CWWKG0016I: Starting server configuration update.
[INFO] [AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /root/workspace/draft-guide-reactive-service-testing/start/system/target/liberty/wlp/usr/servers/defaultServer/configDropins/defaults/system-test-vars.xml
[INFO] [AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /root/workspace/draft-guide-reactive-service-testing/start/system/target/liberty/wlp/usr/servers/defaultServer/configDropins/overrides/liberty-plugin-variable-config.xml
[INFO] [AUDIT   ] CWWKG0017I: The server configuration was successfully updated in 0.328 seconds.
37066 INFO org.microshed.testing.testcontainers.config.TestcontainersConfiguration  - All containers started in 29219ms
37349 INFO org.apache.kafka.clients.consumer.ConsumerConfig  - ConsumerConfig values: 
	allow.auto.create.topics = true
	auto.commit.interval.ms = 5000
	auto.offset.reset = earliest
	bootstrap.servers = [PLAINTEXT://localhost:9093]
	check.crcs = true
	client.dns.lookup = default
	client.id = 
	client.rack = 
	connections.max.idle.ms = 540000
	default.api.timeout.ms = 60000
	enable.auto.commit = true
	exclude.internal.topics = true
	fetch.max.bytes = 52428800
	fetch.max.wait.ms = 500
	fetch.min.bytes = 1
	group.id = system-load-status
	group.instance.id = null
	heartbeat.interval.ms = 3000
	interceptor.classes = []
	internal.leave.group.on.close = true
	isolation.level = read_uncommitted
	key.deserializer = class org.apache.kafka.common.serialization.StringDeserializer
	max.partition.fetch.bytes = 1048576
	max.poll.interval.ms = 300000
	max.poll.records = 500
	metadata.max.age.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.recording.level = INFO
	metrics.sample.window.ms = 30000
	partition.assignment.strategy = [class org.apache.kafka.clients.consumer.RangeAssignor]
	receive.buffer.bytes = 65536
	reconnect.backoff.max.ms = 1000
	reconnect.backoff.ms = 50
	request.timeout.ms = 30000
	retry.backoff.ms = 100
	sasl.client.callback.handler.class = null
	sasl.jaas.config = null
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.login.callback.handler.class = null
	sasl.login.class = null
	sasl.login.refresh.buffer.seconds = 300
	sasl.login.refresh.min.period.seconds = 60
	sasl.login.refresh.window.factor = 0.8
	sasl.login.refresh.window.jitter = 0.05
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	security.providers = null
	send.buffer.bytes = 131072
	session.timeout.ms = 10000
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
	ssl.endpoint.identification.algorithm = https
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLS
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS
	value.deserializer = class io.openliberty.guides.models.SystemLoad$SystemLoadDeserializer

38239 INFO org.apache.kafka.common.utils.AppInfoParser  - Kafka version: 2.4.0
38240 INFO org.apache.kafka.common.utils.AppInfoParser  - Kafka commitId: 77a89fcf8d7fa018
38240 INFO org.apache.kafka.common.utils.AppInfoParser  - Kafka startTimeMs: 1586795324459
38249 INFO org.apache.kafka.clients.consumer.KafkaConsumer  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Subscribed to topic(s): systemLoadTopic
39743 WARN org.apache.kafka.clients.NetworkClient  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Error while fetching metadata with correlation id 3 : {systemLoadTopic=LEADER_NOT_AVAILABLE}
39750 INFO org.apache.kafka.clients.Metadata  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Cluster ID: OAy_HpM-Rs-jMo-GvK5wYg
39978 INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Discovered group coordinator localhost:9093 (id: 2147483646 rack: null)
40000 INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] (Re-)joining group
40107 INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] (Re-)joining group
40191 INFO org.apache.kafka.clients.consumer.internals.ConsumerCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Finished assignment for group at generation 1: {consumer-system-load-status-1-176d71f0-d71a-4401-b852-ec74f59dcb9f=org.apache.kafka.clients.consumer.ConsumerPartitionAssignor$Assignment@62ece7c3}
40282 INFO org.apache.kafka.clients.consumer.internals.AbstractCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Successfully joined group with generation 1
40298 INFO org.apache.kafka.clients.consumer.internals.ConsumerCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Adding newly assigned partitions: systemLoadTopic-0
40383 INFO org.apache.kafka.clients.consumer.internals.ConsumerCoordinator  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Found no committed offset for partition systemLoadTopic-0
40460 INFO org.apache.kafka.clients.consumer.internals.SubscriptionState  - [Consumer clientId=consumer-system-load-status-1, groupId=system-load-status] Resetting offset for partition systemLoadTopic-0 to offset 0.
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
Polled 0 records from Kafka:
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 69.259 s <<< FAILURE! - in it.io.openliberty.guides.system.SystemServiceIT
[ERROR] testCpuStatus  Time elapsed: 30.522 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: No records processed ==> expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:193)
	at it.io.openliberty.guides.system.SystemServiceIT.testCpuStatus(SystemServiceIT.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:675)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:125)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:132)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:124)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:74)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$153/0000000000000000.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall$$Lambda$154/0000000000000000.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$$Lambda$284/0000000000000000.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:104)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:62)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:43)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:35)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:202)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$315/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:198)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$234/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$233/0000000000000000.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$232/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$238/0000000000000000.accept(Unknown Source)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$234/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$233/0000000000000000.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$232/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$238/0000000000000000.accept(Unknown Source)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$234/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$233/0000000000000000.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$232/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$203/0000000000000000.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   SystemServiceIT.testCpuStatus:54 No records processed ==> expected: <true> but was: <false>
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[ERROR] Integration tests failed: There are test failures.

Please refer to /root/workspace/draft-guide-reactive-service-testing/start/system/target/failsafe-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
```

When I try `mvn verify` from `start/system`, it does not build.  When I run `mvn verify` from `finish`, the app builds and the tests pass. Logs:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< io.openliberty.guides:system >--------------------
[INFO] Building system 1.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ system ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ system ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /root/workspace/draft-guide-reactive-service-testing/start/system/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /root/workspace/draft-guide-reactive-service-testing/start/system/src/main/java/io/openliberty/guides/system/SystemService.java:[51,35] package osMean does not exist
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.629 s
[INFO] Finished at: 2020-04-13T09:31:16-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project system: Compilation failure
[ERROR] /root/workspace/draft-guide-reactive-service-testing/start/system/src/main/java/io/openliberty/guides/system/SystemService.java:[51,35] package osMean does not exist
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

## Testing using the Kafka consumer client
### Testing your containers
- Consider adding a
> Navigate to the `start/inventory` directory.

--- a/.gitignore
+++ b/.gitignore
@@ -4,5 +4,5 @@ target/
 .idea/
 .settings
 .vscode
-.project
 .classpath
+.project

--- a/scripts/travisTest.sh
+++ b/scripts/travisTest.sh
@@ -15,7 +15,7 @@ mvn -pl inventory verify
 ./scripts/buildImages.sh
 ./scripts/startContainers.sh
 
-sleep 120
+sleep 150
 
 systemCPULoad="$(curl --write-out "%{http_code}" --silent --output /dev/null "http://localhost:9085/inventory/systems")"
 

--- a/start/system/src/main/java/io/openliberty/guides/system/SystemService.java
+++ b/start/system/src/main/java/io/openliberty/guides/system/SystemService.java
@@ -48,7 +48,7 @@ public class SystemService {
     public Publisher<SystemLoad> sendSystemLoad() {
         return Flowable.interval(15, TimeUnit.SECONDS)
                 .map((interval -> new SystemLoad(getHostname(),
-                        new osMean.getSystemLoadAverage())));
+                        osMean.getSystemLoadAverage())));
     }
     
 }

